src/main/java/billgates/usecases/insert_entry/InsertEntryUseCase.java

```diff
@@ -0,0 +1,32 @@
+package billgates.usecases.insert_entry;
+
+import billgates.database.QueryEntryData;
+import billgates.entities.Entry;
+import billgates.entities.User;
+import billgates.interface_adapters.DatabaseGateway;
+import billgates.usecases.bill_update.BillUpdateUseCase;
+
+import java.util.List;
+
+public class InsertEntryUseCase implements InsertEntryInputPort{
+
+    private final InsertEntryOutputPort presenter;
+    private final DatabaseGateway gateway;
+    private final BillUpdateUseCase update;
+
+    public InsertEntryUseCase(InsertEntryOutputPort presenter, DatabaseGateway gateway, BillUpdateUseCase update){
+        this.presenter = presenter;
+        this.gateway = gateway;
+        this.update = update;
+    }
+
+    @Override
+    public void insertEntry(QueryEntryData entry) {
+        gateway.insertEntry(entry.getId(), entry);
+        update.updateBill();
+        List<Entry> result = this.gateway.getBillData(User.getInstance().getCurrentBillID()).getEntries()
```

Comment:
you may want to remove every line below and including this line in this file and view-model, presenter, and output port to remove the presenter function which is not necessary for this use-case.
once again, you did very good job!